### ISSUE TITLE
Add Material 3 ToggleSwitch Styles

### ISSUE THREAD

**Issue by kazo0** [unoplatform/Uno.Themes#806]
<img width="164" alt="image" src="https://user-images.githubusercontent.com/4793020/172169849-1ca8edd6-28e3-4bb3-a44b-f67bcbc042d4.png">

https://m3.material.io/components/switch/overview

**Comment by agneszitte-nventive** [comment]
Waiting for proper specs and guidelines as of now until I can start creating the new styles

**Comment by agneszitte-nventive** [comment]
Checking with @NVLudwig and @kazo0 if there is still some work to do for this item

**Comment by kazo0** [comment]
I don't think any work has been started on the Themes side of things for M3 switches, unless I am forgetting about it

**Comment by kazo0** [comment]
@NVLudwig is there a branch of the Figma file that has the new ToggleSwitch styles?

**Comment by eriklimakc** [comment]
Should we create a new `ToggleSwitchExtensions` to handle the Icons with DP?

![image](https://github.com/unoplatform/Uno.Themes/assets/114886335/34775aa0-52e7-4b39-b39b-0ec1cd60c366)


It should have `EnabledIcon` and `DisabledIcon`. Maybe `EnableIconVisible` and `DisableIconVisible`?

@kazo0 @agneszitte @Xiaoy312

**Comment by francoistanguay** [comment]
We already ghave an Icon in ControlExtensions... And this is lower priority for me unless Steve thinks we should tackle it now.

**Comment by kazo0** [comment]
Would be good to get this unblocked so we can close out M3 ToggleSwitches

@eriklimakc
Perhaps we can just use the [`OnContent`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.toggleswitch.oncontent?view=winrt-22621) and [`OffContent`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.toggleswitch.offcontent?view=winrt-22621) from the WinUI ToggleSwitch API for this purpose. Instead of it showing as a label next to the switch, you can set it as content that will display within the thumb for the Material style. 

Would be odd looking if someone tried to set it to text but we can document the difference in usage 

Thoughts @agneszitte / @Xiaoy312?

If we can't decide on something now that wouldn't require more work then let's just not deal with icons until another time

**Comment by eriklimakc** [comment]
It works in my tests, but when `OnContent` and/or `OffContent` are not set, the default values are 'On' and 'Off' respectively. To avoid displaying 'On' and 'Off' in the thumb, we may need a converter.

@kazo0 @agneszitte @Xiaoy312

**Comment by eriklimakc** [comment]
It works in my tests, but when OnContent and/or OffContent are not set, the default values are 'On' and 'Off' respectively. To avoid displaying 'On' and 'Off' in the thumb, we may need a converter.


https://github.com/unoplatform/Uno.Themes/assets/114886335/0af0d424-f7e8-4ce1-95c7-ec200b19263e


I'm getting an exception on Windows bc of icons, though... i'm investigating,

@kazo0 @agneszitte @Xiaoy312

**Comment by Xiaoy312** [comment]
as for replacing the original functionality...
it wasnt particularly useful to have "on" and "off" text beside the toggle
not denying it is use, but it is not something hard to do with binding+converter on the users side
plus they would have more control that ways for the text placement + font, and all.
